images/hub/cull_idle_servers.py

```diff
@@ -70,7 +70,13 @@ def cull_one(user, last_activity):
                 method='DELETE',
                 headers=auth_header,
             )
-            yield client.fetch(req)
+            resp = yield client.fetch(req)
+            if resp.code == 202:
+                msg = "Server for {} is slow to stop.".format(user['name'])
+                if cull_users:
+                    app_log.warning(msg + " Not culling user yet.")
+                    return
```

Comment:
Note: this is the magic line that avoid culling the user if the server is still shutting down. It took me two reads to spot this, so leaving a note :) maybe add a comment for future Tims that will overlook this again.

Reply:
comment added for this and above